File: 2160-minimum-operations-to-make-a-uni-value-grid/minimum-operations-to-make-a-uni-value-grid.py
class Solution:
    def minOperations(self, grid: List[List[int]], x: int) -> int:
        nums = sorted([val for row in grid for val in row])
        mod = nums[0] % x

        if any(num % x != mod for num in nums):
            return -1

        nums.sort()
        median = nums[len(nums) // 2]

        count = sum(abs(num - median) // x for num in nums)

        return count

File: 2160-minimum-operations-to-make-a-uni-value-grid/minimum-operations-to-make-a-uni-value-grid.js
/**
 * @param {number[][]} grid
 * @param {number} x
 * @return {number}
 */
var minOperations = function (grid, x) {
    let nums = [];

    for (let row of grid) {
        for (let val of row) {
            nums.push(val);
        }
    }

    let rem = nums[0] % x;

    for (let num of nums) {
        if (num % x !== rem) {
            return -1;
        }
    }

    nums.sort((a, b) => a - b);

    let median = nums[Math.floor(nums.length / 2)];

    let ops = 0;

    for (let num of nums) {
        ops += Math.abs(num - median) / x;
    }

    return ops;
};